cmd/rule-evaluator/main.go

```diff
@@ -569,6 +581,34 @@ func sendAlerts(s *notifier.Manager, externalURL *url.URL) rules.NotifyFunc {
 	}
 }
 
+func googleCloudLink(projectID, expr string, endTime, startTime time.Time) string {
+	// Clone URL to avoid mutating the original.
+	url := googleCloudBaseURL
+	url.Path += ";endTime=" + endTime.Format(time.RFC3339)
+	url.Path += ";startTime=" + startTime.Format(time.RFC3339)
+
+	// These settings reflect the default on metrics explorer for majority of use-cases.
+	pageState := map[string]any{
+		"xyChart": map[string]any{
+			"dataSets": []map[string]any{
+				{
+					"prometheusQuery": expr,
```

Comment:
question: Does `expr` need to be [escaped](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/main/vendor/github.com/prometheus/prometheus/util/strutil/strconv.go#L29)?

Reply:
Yes. Using `json.Marshal` escapes the JSON quotes. Then, we use `Query().Encode()`, which encodes the JSON within the URL query.

Reply:
SGTM